test/system/253-podman-quadlet.bats

```diff
@@ -0,0 +1,323 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# Tests generated configurations for systemd.
+#
+
+# bats file_tags=ci:parallel
+
+load helpers
+load helpers.network
+load helpers.registry
+load helpers.systemd
+
+UNIT_FILES=()
+
+function start_time() {
+    sleep_to_next_second # Ensure we're on a new second with no previous logging
+    STARTED_TIME=$(date "+%F %R:%S") # Start time for new log time
+}
+
+function setup() {
+    skip_if_remote "quadlet tests are meaningless over remote"
+    skip_if_rootless_cgroupsv1 "Can't use --cgroups=split w/ CGv1 (issue 17456, wontfix)"
+    skip_if_journald_unavailable "Needed for RHEL. FIXME: we might be able to re-enable a subset of tests."
+
+    test -x "$QUADLET" || die "Cannot run quadlet tests without executable \$QUADLET ($QUADLET)"
+
+    start_time
+
+    basic_setup
+}
+
+function teardown() {
+    for UNIT_FILE in ${UNIT_FILES[@]}; do
+        if [[ -e "$UNIT_FILE" ]]; then
+            local service=$(basename "$UNIT_FILE")
+            run systemctl stop "$service"
+            if [ $status -ne 0 ]; then
+               echo "# WARNING: systemctl stop failed in teardown: $output" >&3
+            fi
+            run systemctl reset-failed "$service"
+            rm -f "$UNIT_FILE"
+        fi
+    done
+    systemctl daemon-reload
+
+    basic_teardown
+}
+
+@test "quadlet verb - install, list, rm" {
+    # Create a test quadlet file
+    local quadlet_file=$PODMAN_TMPDIR/alpine-quadlet.container
+    cat > $quadlet_file <<EOF
+[Container]
+Image=$IMAGE
+Exec=sh -c "echo STARTED CONTAINER; trap 'exit' SIGTERM; while :; do sleep 0.1; done"
+Notify=yes
+LogDriver=passthrough
+EOF
+    # Clean all quadlets
+    run_podman quadlet rm --all -f
+
+    # Test quadlet install
+    run_podman quadlet install $quadlet_file
+
+    # Test quadlet list
+    run_podman quadlet list
+    assert "$output" =~ "alpine-quadlet.container" "list should contain alpine-quadlet.container"
+
+    # Test quadlet list with filter
+    run_podman quadlet list --filter name=something*
+    assert "$output" !~ "alpine-quadlet.container" "filtered list should not contain alpine-quadlet.container"
+
+    # Test quadlet list with matching filter
+    run_podman quadlet list --filter name=alpine*
+    assert "$output" =~ "alpine-quadlet.container" "matching filter should contain alpine-quadlet.container"
+
+    # Test quadlet print
+    run_podman quadlet print alpine-quadlet.container
+    assert "$output" =~ "\[Container\]" "print should show container section"
+    assert "$output" =~ "Image=$IMAGE" "print should show correct image"
+
+    # Test quadlet rm
+    run_podman quadlet rm alpine-quadlet.container
+
+    # Verify removal
+    run_podman quadlet list
+    assert "$output" !~ "alpine-quadlet.container" "list should not contain removed container"
+}
+
+# bats test_tags=distro-integration
```

Comment:
please drop these tags, nothing in these tests depends in any way on an actual external dependency so there is no real reason to specially run them in our fedora openQA env.